nvtabular/workflow.py

```diff
@@ -296,35 +296,46 @@ def _get_stat_ops(nodes):
     return set(node for node in iter_nodes(nodes) if isinstance(node.op, StatOperator))
 
 
-def _transform_partition(root_gdf, column_groups):
+def _get_unique(cols):
+    # Need to preserve order in unique-column list
+    unique_cols = []
+    for c in cols:
+        if c not in unique_cols:
+            unique_cols.append(c)
+    return unique_cols
```

Comment:
Its sorta annoying that sets in python aren't ordered - but since dict's are ordered in python, maybe we can take advantage of that ? 

```suggestion
def _get_unique(cols):
    # Need to preserve order in unique-column list
    return list({x: x for x in cols}.keys())
```

Reply:
also - I think we should probably move this 'unique' functionality to the column_group class (inside flattened_columns rather than here)

Reply:
>dict's are ordered in python

I didn't realize dicts are ordered?

Reply:
yup! since python 3.7 dictionary order is defined to be the same as insertion order: https://softwaremaniacs.org/blog/2020/02/05/dicts-ordered/en/ or https://mail.python.org/pipermail/python-dev/2017-December/151283.html

Reply:
Nice! I remember finding a serious bug related to a dict **not** being ordered back in the <=3.5 days.  Good to know :)